src/Fixer/Whitespace/ArrayIndentationFixer.php

```diff
@@ -0,0 +1,331 @@
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should run after BracesFixer
+        return -30;
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        foreach ($this->findArrays($tokens) as $array) {
+            $previousLineBracesDelta = 1;
+            $indentLevel = 0;
+            $ignoredPositiveDelta = 0;
+            $arrayIndent = $this->getLineIndentation($tokens, $array['start']);
+            $previousLineInitialIndent = $arrayIndent;
+            $previousLineNewIndent = $arrayIndent;
+
+            foreach ($array['line_braces_deltas'] as $index => $lineBracesDelta) {
+                $token = $tokens[$index];
+
+                $extraIndentLevel = 0;
+
+                if ($previousLineBracesDelta > 0) {
+                    ++$indentLevel;
+
+                    if ($previousLineBracesDelta > 1) {
+                        $ignoredPositiveDelta += $previousLineBracesDelta - 1;
+                    }
+                }
+
+                if ($lineBracesDelta < 0) {
+                    $delta = $lineBracesDelta + $ignoredPositiveDelta;
+                    $ignoredPositiveDelta = max(0, $ignoredPositiveDelta + $lineBracesDelta);
+                    $indentLevel = max(0, $indentLevel + $delta);
+
+                    if ($this->isClosingLineWithMeaningfulContent($tokens, $index)) {
+                        $extraIndentLevel = -$lineBracesDelta;
+                    }
+                }
+
+                if ($this->newlineIsInArrayScope($tokens, $index, $array)) {
+                    $content = preg_replace(
+                        '/(\R)[\t ]*$/',
+                        '$1'.$arrayIndent.str_repeat($this->whitespacesConfig->getIndent(), $indentLevel + $extraIndentLevel),
+                        $token->getContent()
+                    );
+
+                    $previousLineInitialIndent = $this->extractIndent($token->getContent());
+                    $previousLineNewIndent = $this->extractIndent($content);
+                } else {
+                    $content = preg_replace(
+                        '/(\R)'.preg_quote($previousLineInitialIndent, '/').'([\t ]*)$/',
+                        '$1'.$previousLineNewIndent.'$2',
+                        $token->getContent()
+                    );
+                }
+
+                $previousLineBracesDelta = $lineBracesDelta;
+
+                $tokens[$index] = new Token([T_WHITESPACE, $content]);
+            }
+        }
+    }
+
+    private function findArrays(Tokens $tokens)
+    {
+        $arrays = [];
+
+        foreach ($this->findArrayTokenRanges($tokens, 0, count($tokens) - 1) as $arrayTokenRanges) {
+            $arrays[] = [
+                'start' => $arrayTokenRanges[0][0],
+                'end' => $arrayTokenRanges[count($arrayTokenRanges) - 1][1],
+                'token_ranges' => $arrayTokenRanges,
+            ];
```

Comment:
maybe
```php

// remove $arrays

foreach(...) {
    $array = [
        'start' => $arrayTokenRanges[0][0],
        'end' => $arrayTokenRanges[count($arrayTokenRanges) - 1][1],
        'token_ranges' => $arrayTokenRanges,
    ];
    $array['line_braces_deltas'] = $this->computeArrayLineBracesDeltas($tokens, $array);

    yield $array;
}

// remove next foreach and return statement
```

Reply:
I removed the second `foreach` but didn't use `yield`.